lmnet/lmnet/datasets/camvid.py

```diff
@@ -140,15 +201,94 @@ def files_and_annotations(self):
         image_files = [filename.replace("/SegNet/CamVid", self.data_dir) for filename in image_files]
         label_files = [filename.replace("/SegNet/CamVid", self.data_dir) for filename in label_files]
 
+        image_files, label_files = shuffle(image_files, label_files)
+        print("files and annotations are ready")
+
         return image_files, label_files
 
-    def __getitem__(self, i, type=None):
-        image_files, label_files = self.files_and_annotations
 
-        image = get_image(image_files[i])
-        label = get_image(label_files[i], convert_rgb=False).copy()
+class CamvidCustom(StoragePathCustomizable, CamvidBase):
+    """CamvidCustom
 
-        return (image, label)
+    CamVid base custom dataset format.
+    """
 
-    def __len__(self):
-        return self.num_per_epoch
+    def __init__(
+            self,
+            batch_size=10,
+            *args,
+            **kwargs
+    ):
+
+        super().__init__(
+            batch_size=batch_size,
+            *args,
+            **kwargs,
+        )
+        self.parse_labels()
+
+    @property
+    def label_colors(self):
+        classes = self.parse_labels()
+        return COLOR_MAP[:len(classes)]
```

Comment:
If the length of class is over than 21, the label_colors() return 20 length list of colors, I think. Is the length of colors what you expect ??

Reply:
@ruimashita Oh, this is my lack of consideration.
I added `get_color_map` function to get arbitrary length color map.
I also added unit test in `test_common.py`

Thanks!!